### PR TITLE
Add more unit tests

### PR DESCRIPTION
**Pull request by jim-minter**

### CODE REVIEW

File: pkg/util/subnet/subnet_test.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+	mgmtnetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-07-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/golang/mock/gomock"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	mock_network "github.com/Azure/ARO-RP/pkg/util/mocks/azureclient/mgmt/network"
+)
+
+func TestGet(t *testing.T) {
```
Comment (m1kola) [inline pkg/util/subnet/subnet_test.go]: It looks like we are testing `Split` and `azure.ParseResourceID` in `Get` and `CreateOrUpdate` tests and that the `manager` itself has no logic. Is it worth having these tests?

File: pkg/util/subnet/subnet_test.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+	for _, tt := range []struct {
+		name      string
+		subnetID  string
+		wantNSGID string
+		wantErr   string
+	}{
+		{
+			name:      "master",
+			subnetID:  "/subscriptions/subscriptionId/resourceGroups/vnetResourceGroup/providers/Microsoft.Network/virtualNetworks/vnet/subnets/master",
+			wantNSGID: "/subscriptions/subscriptionId/resourceGroups/clusterResourceGroup/providers/Microsoft.Network/networkSecurityGroups/aro-controlplane-nsg",
```
Comment (m1kola) [inline pkg/util/subnet/subnet_test.go]: I noticed that we've got has NSG names hardcoded in `subnet` and in `installed`. Our current tests will not detect the fact that we updated it in one place and failed to do so in other places.

Should we create a medium priority issue to review codebase and replace values like this with constants and\or add tests to cover this?

File: pkg/util/dns/dns_test.go [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+		wantErr string
+	}
+
+	for _, tt := range []*test{
+		{
+			name: "managed, new record",
+			oc:   managedOc,
+			mocks: func(tt *test, recordsets *mock_dns.MockRecordSetsClient) {
+				recordsets.EXPECT().
+					Get(ctx, "rpResourcegroup", "test", "api.domain", mgmtdns.A).
```
[the remark this answers is not in the document]
Comment (jim-minter) [reply pkg/util/dns/dns_test.go]: I agree with the sentiment; I wonder if now is the time to make env/test.go just a dummy struct with fields that can be manipulated, and set those fields from each unit test that uses env.Test.

File: pkg/util/subnet/generate.go [hunk 4]
```diff
@@ -1,7 +0,0 @@
-package subnet
-
-// Copyright (c) Microsoft Corporation.
-// Licensed under the Apache License 2.0.
-
-//go:generate go run ../../../vendor/github.com/golang/mock/mockgen -destination=../mocks/$GOPACKAGE/$GOPACKAGE.go github.com/Azure/ARO-RP/pkg/util/$GOPACKAGE Manager
```
[the remark this answers is not in the document]
Comment (jim-minter) [reply pkg/util/subnet/generate.go]: it happens to be in disks.go because it's alphabetically the first non _addons.go.  Not a good reason ;-).  Yeah, happy to revert this change and if you want to pull them all out into generate.go files, do.

File: pkg/util/subnet/subnet_test.go (hunk 1 above)
Comment (jim-minter) [reply pkg/util/subnet/subnet_test.go]: It's borderline, isn't it.

File: pkg/util/subnet/subnet_test.go (hunk 2 above)
Comment (jim-minter) [reply pkg/util/subnet/subnet_test.go]: Yes.  A whole pile of these constants are implicitly mandated by the installer, or worse, cluster-api-provider-azure.

File: pkg/util/dns/dns_test.go [hunk 5]
```diff
@@ -0,0 +1,473 @@
...
+		},
+		{
+			domain: "foo.test",
+			want:   "foo",
+		},
+		{
+			domain: "foo.other",
+			want:   "",
+		},
+	} {
```
[the remark this answers is not in the document]
Comment (jim-minter) [reply pkg/util/dns/dns_test.go]: I'm not sure what the best thing to do is.  At https://github.com/Azure/ARO-RP/pull/49/files#diff-dbeb1a47e07b6d03f92335819e5015c1R252-R259 this logic becomes more dependent on receiving valid input.  I wonder if in #49 we should play it safe, revalidate against the edge cases and return an error if invalid ("error: this should never happen") - at that point unit test coverage would definitely make sense.

[the remark this answers is not in the document]
Comment (m1kola) [reply pkg/util/dns/dns_test.go]: I like "error: this should never happen"-way. Explicit is better than implicit (excuse my Python).

File: pkg/util/subnet/generate.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (m1kola) [reply pkg/util/subnet/generate.go]: Created https://github.com/Azure/ARO-RP/issues/71

File: pkg/util/dns/dns_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (jim-minter) [reply pkg/util/dns/dns_test.go]: pushed to #49

File: pkg/util/subnet/subnet_test.go (hunk 1 above)
Comment (jim-minter) [reply pkg/util/subnet/subnet_test.go]: I think I'm going to leave these in to encourage them to grow if the manager functionality changes/is added to.

**Comment by jim-minter** [comment]
@m1kola, all comments addressed - sign-off would be appreciated, thanks!
